drivers/led/led_gpio.c

```diff
@@ -60,17 +93,20 @@ static int led_gpio_init(const struct device *dev)
 	}
 
 	for (size_t i = 0; (i < config->num_leds) && !err; i++) {
-		const struct gpio_dt_spec *led = &config->led[i];
-
-		if (device_is_ready(led->port)) {
-			err = gpio_pin_configure_dt(led, GPIO_OUTPUT_INACTIVE);
-
-			if (err) {
-				LOG_ERR("Cannot configure GPIO (err %d)", err);
+		const struct led_gpio *led = &config->led[i];
+
+		for (size_t j = 0; (j < led->num_gpios) && !err; j++) {
+			const struct gpio_dt_spec *gpio = &led->gpio[j];
+
+			if (device_is_ready(gpio->port)) {
+				err = gpio_pin_configure_dt(gpio, GPIO_OUTPUT_INACTIVE);
+				if (err) {
+					LOG_ERR("Cannot configure GPIO (err %d)", err);
```

Comment:
Shouldn't we exit with errno in that case as well?

Reply:
We do. `err` non null is an exit condition in the `for` loops above.

Reply:
Ops, I didn't notice that.